src/test/scala/chiselTests/WidthSpec.scala

```diff
@@ -42,6 +44,30 @@ class WidthSpec extends ChiselFlatSpec {
     0.U.getWidth should be(1)
     0.S.getWidth should be(1)
   }
+
+  behavior.of("Empty Aggregates")
+
+  they should "have a width of 0" in {
+    assertKnownWidth(0) {
+      Wire(Vec(0, UInt(8.W)))
+    }
+    assertKnownWidth(0) {
+      Wire(new EmptyBundle)
+    }
+  }
+
+  // This is a bug that has existed for basically forever
+  // This really should be assertKnownWidth(0)
+  they should "result in a 1-bit UInt when calling .asUInt" in {
+    assertInferredWidth(1) {
+      val x = Wire(Vec(0, UInt(8.W))).asUInt
+      WireInit(x)
+    }
+    assertInferredWidth(1) {
+      val x = Wire(new EmptyBundle).asUInt
+      WireInit(x)
+    }
+  }
```

Comment:
My original change resulted in some surprising test failures and exposed some concerning buggy behavior. As shown here, if you call `.asUInt` on an empty Aggregate, you get `0.U` which is `0.U(1.W)`. Fortunately, when these empty Aggregates are part of larger Aggregates, they are flattened out (as zero-width wires), but in the special case of just `.asUInt` on a empty Aggregate, this is what happens.

This is definitely a bug, obviously it should hold that `x.getWidth == x.asUInt.getWidth`, but such width issues are tricky to change. This one derives from calling `SeqUtils.asUInt` which is the same reason why `Cat(Seq())` returns `0.U` (which again is `0.U(1.W)`.

I think it's reasonable to fix this particular bug while leaving the `Cat` and other similar cases alone, but such changes are a little sketchy because it can result in width changes for downstream things and thus silently change behavior. In particular, it's scary if generator code is concatenating the result of `.asUInt` on a `Vec` that can be parameterized all the way down to size 0, or similarly with a Record or Bundle that could be empty.

Thoughts @aswaterman @azidar @seldridge @mwachs5?

Reply:
In any case, this PR is not really trying to be concerned with this change and just documents and maintains the existing behavior.

Reply:
> obviously it should hold that `x.getWidth == x.asUInt.getWidth`

I think this is the overriding concern and fixing it is worth the pain.